.github/workflows/build-test.yaml

```diff
@@ -43,6 +43,8 @@ jobs:
         with:
           go-version: "${{ env.GO_VERSION }}"
           cache: "true"
+      - name: "go mod download" # prevent CI cache poisoning by getting all deps
```

Comment:
How, exactly, does this not poison the cache?

Reply:
go build omits all dependencies used for testing. Go mod download downloads all the modules. By calling it before go build, we force all modules to be downloaded.